internal/objects/validator.go

```diff
@@ -54,6 +54,15 @@ type request struct {
 	obj    *unstructured.Unstructured
 	oldObj *unstructured.Unstructured
 	op     k8sadm.Operation
+	gvr    metav1.GroupVersionResource
+}
+
+func (req request) GroupResource() schema.GroupResource {
+	return schema.GroupResource{Group: req.gvr.Group, Resource: req.gvr.Resource}
+}
+
+func (req request) NamespacedName() types.NamespacedName {
+	return types.NamespacedName{Namespace: req.obj.GetNamespace(), Name: req.obj.GetName()}
```

Comment:
I'd optimize these names for conciseness since they're internal to this file, e.g.

```
func (r request) gr() schema.GroupResource() {...}
func (r request) name() string {return ...}
```

This will make the boilerplate code much easier to read, e.g.

`return webhooks.DenyConflict(req.gr(), req.name(), err)`

instead of

`return webhooks.DenyConflict(req.GroupResource(), req.NamespacedName().String(), err)`

In addition, lowercase names on the functions will clearly indicate to the reader that these are local to this package. I often find that exported names on an unexported type like `req` are always a bit confusing to the reader.

Reply:
Agreed and fixed! PTAL!